docs/develprocess.md

```diff
@@ -182,6 +182,22 @@ Special case: if the upstream version and the XCP-ng version are always the same
 * If we get the sources from XS SRPMs, then we import them to a branch named `XS` and tag `XS-XSVERSIONFULL` (`XS-8.2.0`). Example: `host-installer`.
 * If we forked a git repository, we don't need to push the upstream branches or tags to our own fork. However it could be a good habit to track maintenance or hotfix branches for changes.
 
+#### Special case: `sm`
+
+##### How to create a new branch after a XCP-ng release?
+
+- Find the tag of the new sm release in the upstream repo and checkout it.
+- Create a new 8.x branch.
+- Cherry-pick our patches (of the previous maintenance 8.(x-1) branch) and resolve conflicts.
+
+##### How to synchronize a maintenance branch with the upstream?
+
+- First, it's necessary to checkout the latest current upstream commit of the maintenance branch, so just before our own changes.
+- Then create a new maintenance branch, and cherry-pick the updates using the SRPM changelog of the upstream version.
+- Check if we have the same source code between the SRPM and our branch using `diff -urq <sources> <upstream sources>`.
+- Because the upstream sm doesn't provide maintenance updates, we must tag manually the last upstream commit. (This tag will be used to checkout for the next maintenance update.)
+- After that, it's necessary to cherry-pick our patches and resolve conflicts.
```

Comment:
```suggestion
The upstream repository doesn't contain the maintenance branches that are the basis for Citrix Hypervisor hotfixes. So the process is a bit more complicated for us, as we first need to create a branch whose contents matches the contents of the tar.gz in the hotfix SRPM, and tag this as a reference for generating our patches for the RPM.

- Create a new maintenance branch, named following the `UPSTREAMVERSION-XCPNGVERSION` convention, from the tag that matches the current (before the hotfix) version of `sm` in our RPM.
  - If it's the first `sm` hotfix for this release of XCP-ng, then start from the upstream tag
  - Else start from the tag that we had to create for the previous hotfix because upstream didn't provide one. Named following a `vUPSTREAMVERSION-xcpng` convention. Example: `v2.30.4-xcpng`.
- Cherry-pick upstream commits into this branch using the SRPM changelog of the upstream hotfix.
- Check if we have the same source code between the SRPM and our branch using `diff -urq <sources> <upstream sources>`.
- When our branch matches exactly the contents of the hotfix's tarball, tag it as `vUPSTREAMVERSION-xcpng`. This tag will be used to checkout for the next maintenance update, and as the base reference to generate patches for our RPM.
- We can at last apply our specific commits on top: rebase or cherry-pick them from the previous maintenance branch for this release (example: `2.30.3-8.2`) and resolve conflicts carefully. Of course, drop patches that were already merged upstream.
```